### ISSUE TITLE
Handling of non-words in publisher query

### ISSUE THREAD

**Issue by kelson42** [openzim/libzim#537]
The approach used in `publisherQuery()` for building a phrase query
enforcing the specified prefix for all terms fails if the input phrase
contains a term that Xapian's query parser doesn't like (e.g. a
standalone ampersand character, 1/2, a#1, etc).

Using the `quest` tool (coming with xapian-tools under Ubuntu) the
issue can be demonstrated as follows:

```
$ quest -o phrase -d some_xapian_db "Energy & security act"
UnimplementedError: OP_NEAR and OP_PHRASE only currently support leaf subqueries

$ quest -o phrase -d some_xapian_db 'Energy 1/2 security act'
UnimplementedError: OP_NEAR and OP_PHRASE only currently support leaf subqueries

$ quest -o phrase -d some_xapian_db "Energy a#1 security act"
UnimplementedError: OP_NEAR and OP_PHRASE only currently support leaf subqueries
```

The problem comes from parsing the query with the default operation set
to `OP_PHRASE` (exemplified by the `-o phrase` option in above
invocations of `quest`). A workaround is to parse the phrase with a
default operation of `OP_OR` and then combine all the terms with
`OP_PHRASE`.

This issue is likely to be present in libzim/src/search.cpp too, where
`set_default_op(Xapian::Query::op::OP_PHRASE)` is used. It affects
suggestions in my local build of kiwix-serve (though the instance
running at library.kiwix.org seems to be unaffected, is it an old
version?).

**Comment by kelson42** [comment]
Duplicate of #536

**Comment by veloman-yunkan** [comment]
@kelson42 I don't see why you had to open this issue in libzim and then close it as a duplicate for #536. This bug was introduced and then fixed in the yet umerged PR kiwix/kiwix-lib#488. The only relation of that virtual issue to libzim is that an approach from libzim source code was utilized in new code added to kiwix-lib. The only thing that has to be done about that yet unmerged piece of code in kiwix-lib is to incorporate the extra fix (disabling of stemming) that @maneeshpm has done in #536.

**Comment by kelson42** [comment]
@veloman-yunkan I got confused, but to avoid any confusion in the future please always open a bug ticket before a fixing PR.
